packages/next/src/server-build.ts

```diff
@@ -1197,65 +1188,104 @@ export async function serverBuild({
         });
       }
 
-      for (const page of group.pages) {
-        const pageNoExt = page.replace(/\.js$/, '');
-        let isPrerender = prerenderRoutes.has(
-          path.join('/', pageNoExt === 'index' ? '' : pageNoExt)
-        );
+      for (const pageFilename of group.pages) {
+        // This is the name of the page, where the root is `index`.
+        const pageName = pageFilename.replace(/\.js$/, '');
+
+        // This is the name of the page prefixed with a `/`, where the root is
+        // `/index`.
+        const pagePath = path.posix.join('/', pageName);
+
+        // This is the routable pathname for the page, where the root is `/`.
+        const pagePathname = pagePath === '/index' ? '/' : pagePath;
+
+        let isPrerender = prerenderRoutes.has(pagePathname);
 
         if (!isPrerender && routesManifest?.i18n) {
           isPrerender = routesManifest.i18n.locales.some(locale => {
             return prerenderRoutes.has(
-              path.join('/', locale, pageNoExt === 'index' ? '' : pageNoExt)
+              path.join('/', locale, pageName === 'index' ? '' : pageName)
             );
           });
         }
-        let outputName = path.posix.join(entryDirectory, pageNoExt);
 
-        if (!group.isAppRouter && !group.isAppRouteHandler) {
-          outputName = normalizeIndexOutput(outputName, true);
-        }
+        let outputName = path.posix.join(entryDirectory, pageName);
 
         // If this is a PPR page, then we should prefix the output name.
         if (isPPR) {
           if (!revalidate) {
             throw new Error("Invariant: PPR lambda isn't set");
           }
 
-          // Get the get the base path prefixed route, without the index
-          // normalization.
-          outputName = path.posix.join(entryDirectory, pageNoExt);
+          // Assign the revalidate lambda to the output name. That's used to
+          // perform the initial static shell render.
           lambdas[outputName] = revalidate;
 
-          const pprOutputName = path.posix.join(
-            entryDirectory,
-            '/_next/postponed/resume',
-            pageNoExt
-          );
-          lambdas[pprOutputName] = lambda;
+          // If this is an omitted page, them it should be routed to the
+          // specific dynamic route, and not the dynamic one.
```

Comment:
Not sure that I understand what this means ("routed to the dynamic route, and not the dynamic one")